pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatPatternSpec.java

```diff
@@ -56,8 +59,8 @@ public DateTimeFormatPatternSpec(TimeFormat timeFormat) {
   public DateTimeFormatPatternSpec(TimeFormat timeFormat, @Nullable String sdfPatternWithTz) {
     _timeFormat = timeFormat;
     if (timeFormat == TimeFormat.SIMPLE_DATE_FORMAT) {
-      Preconditions.checkArgument(StringUtils.isNotEmpty(sdfPatternWithTz), "Must provide SIMPLE_DATE_FORMAT pattern");
-      Matcher m = SDF_PATTERN_WITH_TIMEZONE.matcher(sdfPatternWithTz);
+      Matcher m = sdfPatternWithTz != null ? SDF_PATTERN_WITH_TIMEZONE.matcher(sdfPatternWithTz)
```

Comment:
We should skip the regex matching if it does not exist
```suggestion
      if (sdfPatternWithTz != null) {
        ...
      } else {
        _sdfPattern = null;
        _dateTimeZone = DEFAULT_DATE_TIME_ZONE;
      }
```